### DISCUSSION TITLE
Fix for Jamulus server and gui crashing on latest opensuse tumbleweed

### DISCUSSION THREAD

**Discussion by joba-1** [jamulussoftware/jamulus#3250] [General]
since approximately start of march 2024 jamulus provided by the opensuse tumbleweed distro crashes when run on my servers and notebooks. The pipewire soundsystem was updated in that timeframe but not the jamulus package. So I guess pipewire came with new libs that are not compatible with the current jamulus build.

Fix is to do a self compile:
```
sudo zypper in qt6-base-common-devel qt6-network-devel qt6-widgets-devel qt6-xml-devel qt6-concurrent-devel qt6-multimedia-devel qt6-tools-linguist
cd
git clone https://github.com/jamulussoftware/jamulus.git
cd jamulus
qmake6
make -j8
sudo make install
```

To update the services to use the self compiled version (if you use them)
```
sudo sed -i s@/usr/bin/Jamulus@/usr/local/bin/Jamulus@ /etc/systemd/system/jamulus-*.service
sudo systemctl daemon-reload
```

What made it difficult (I have no xp with compiling qt apps): 
* The official compile instructions fail because there is no Makefile at first
* Docs talk about QT5 but I have QT6 -> use qmake6

Hope this helps someone. Thanks for the cool software!

**Comment by rdica** [comment]
@joba-1 Hi, glad you were able to figure it out!

Is this the official compile instructions you're referring to https://github.com/jamulussoftware/jamulus/blob/main/COMPILING.md ?

Have you reached out to the maintainer of the jamulus pkg for opensuse to let them know their pkg needs updating?

**Reply by joba-1** [reply]
Q1: yes
Q2: no, I‘ll try to find them

**Reply by joba-1** [reply]
https://bugzilla.opensuse.org/show_bug.cgi?id=1222081

**Reply by rdica** [reply]
Good deal. Here's hoping the maintainer updates the pkg!

**Comment by ann0see** [comment]
I now also got a victim of a segfault myself while connecting to a server. 

```
Thread 8 "Jamulus" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff03296c0 (LWP 6701)]
0x000055555568d99e in ec_enc_shrink ()
```

will debug this later on - but this is **very** odd.

Edit: It's due to the updated opus version I had locally. So known issue which doesn't happen in production.

**Comment by joba-1** [comment]
fwiw, there was a recent update of the tumbleweed package to Jamulus 3.10.0-1.3

It still crashes, so I stay with self compiled…

**Reply by ann0see** [reply]
It could be somewhat related as both our crashes seemed to be related to something with opus. 
Do they use a newer opus version?

**Comment by joba-1** [comment]
hm, there was a new libopus0-1.5.2-1.1.x86_64.rpm from 2 days ago.
Installed it, restarted jamulus service-> crash :(
